modvendor/modvendor_test.go

```diff
@@ -15,18 +15,823 @@
+						Module(
+							Labels("test"),
+							Str("source", "git::file://{{.}}/cool-module?ref=main"),
+						),
+					),
+				},
+				{
+					repo: "cool-module",
+					path: "cool-module/main.tf",
+					data: Doc(
+						Module(
+							Labels("test"),
+							Str("source", "git::file://{{.}}/forgotten-module?ref=main"),
+						),
+					),
+				},
+			},
+			wantVendored: []string{
+				"git::file://{{.}}/my-module?ref=main",
+				"git::file://{{.}}/awesome-module?ref=main",
+				"git::file://{{.}}/cool-module?ref=main",
+				"git::file://{{.}}/best-module?ref=main",
+				"git::file://{{.}}/forgotten-module?ref=main",
+			},
+			wantFiles: map[vendorPathSpec]fmt.Stringer{
+				"git::file://{{.}}/my-module?ref=main#main.tf": Module(
+					Labels("test"),
+					Str("source", "{{index . 1}}"),
+				),
+				"git::file://{{.}}/my-module?ref=main#other.tf": Doc(
+					Module(
+						Labels("test"),
+						Str("source", "{{index . 3}}"),
+						Str("other", "value"),
+					),
+					Module(
+						Labels("test2"),
+						Str("source", "{{index . 1}}"),
+					),
+				),
+				"git::file://{{.}}/awesome-module?ref=main#main.tf": Doc(
+					Module(
+						Labels("test"),
+						Str("source", "{{index . 2}}"),
+					),
+				),
+				"git::file://{{.}}/cool-module?ref=main#main.tf": Doc(
+					Module(
+						Labels("test"),
+						Str("source", "{{index . 4}}"),
+					),
+				),
+			},
+		},
+		{
+			name: "cycle: my-module -> (awesome-module -> cool-module -> my-module, best-module)",
+			layout: []string{
+				"g:my-module",
+				"g:awesome-module",
+				"g:best-module",
+				"g:cool-module",
+			},
+			source: "git::file://{{.}}/my-module?ref=main",
+			configs: []hclconfig{
+				{
+					repo: "my-module",
+					path: "my-module/main.tf",
+					data: Doc(
+						Module(
+							Labels("test"),
+							Str("source", "git::file://{{.}}/awesome-module?ref=main"),
+						),
+					),
+				},
+				{
+					repo: "my-module",
+					path: "my-module/other.tf",
+					data: Doc(
+						Module(
+							Labels("test"),
+							Str("source", "git::file://{{.}}/best-module?ref=main"),
+							Str("other", "value"),
+						),
+						Module(
+							Labels("test2"),
+							Str("source", "git::file://{{.}}/awesome-module?ref=main"),
+						),
+					),
+				},
+				{
+					repo: "awesome-module",
+					path: "awesome-module/main.tf",
+					data: Doc(
+						Module(
+							Labels("test"),
+							Str("source", "git::file://{{.}}/cool-module?ref=main"),
+						),
+					),
+				},
+				{
+					repo: "cool-module",
+					path: "cool-module/main.tf",
+					data: Doc(
+						Module(
+							Labels("test"),
+							Str("source", "git::file://{{.}}/my-module?ref=main"),
+						),
+					),
+				},
+			},
+			wantVendored: []string{
+				"git::file://{{.}}/my-module?ref=main",
+				"git::file://{{.}}/awesome-module?ref=main",
+				"git::file://{{.}}/cool-module?ref=main",
+				"git::file://{{.}}/best-module?ref=main",
+			},
+			wantFiles: map[vendorPathSpec]fmt.Stringer{
```

Comment:
Shouldn't "cool-module" be on the wantFiles list with a rewritten source to my-module ? In case of a cycle I was expecting the module to be vendored + have the source patched too, no ? (sorry if missing something basic here)

Reply:
just made this work. Thanks